disassemblers/ofrak_pyghidra/ofrak_pyghidra/components/pyghidra_components.py

```diff
@@ -0,0 +1,112 @@
+from ofrak.core import *
+from tempfile import TemporaryDirectory
+import os
+
+
+from ofrak_cached_disassembly.components.cached_disassembly import CachedAnalysisStore
+from ofrak_cached_disassembly.components.cached_disassembly_unpacker import (
+    CachedCodeRegionUnpacker,
+    CachedComplexBlockUnpacker,
+    CachedBasicBlockUnpacker,
+    CachedCodeRegionModifier,
+    CachedDecompilationAnalyzer,
+)
+from ofrak_pyghidra.standalone.pyghidra_analysis import unpack
+
+
+_GHIDRA_AUTO_LOADABLE_FORMATS = [Elf, Ihex, Pe]
+
+
+@dataclass
+class PyGhidraAutoLoadProject(ResourceView):
+    pass
+
+
+@dataclass
+class PyGhidraProject(ResourceView):
+    pass
+
+
+class PyGhidraAnalysisIdentifier(Identifier):
+    """
+    Component to identify resources to analyze with Ghidra. If this component is discovered,
+    it will tag all [Program][ofrak.core.program.Program]s as GhidraProjects
+    """
+
+    id = b"GhidraAnalysisIdentifier"
+    targets = (Program,)
+
+    async def identify(self, resource: Resource, config=None):
+        for tag in _GHIDRA_AUTO_LOADABLE_FORMATS:
+            if resource.has_tag(tag):
+                resource.add_tag(PyGhidraAutoLoadProject)
+
+
+@dataclass
+class PyGhidraUnpackerConfig(ComponentConfig):
+    unpack_complex_blocks: bool
+
+
+class PyGhidraAnalysisStore(CachedAnalysisStore):
+    pass
+
+
+class CachedCodeRegionModifier(CachedCodeRegionModifier):
+    pass
```

Comment:
The cached code region unpacker runs this modifier directly, this is a workaround to have it be discoverable with only pyghidra components injected. I tested that it does not cause conflicts if both the cached_disassembly and pyghidra are injected at the same time.